fs/sync/sync.go

```diff
@@ -482,6 +482,50 @@ func deleteEmptyDirectories(f fs.Fs, entries fs.DirEntries) error {
 	return nil
 }
 
+// This copies the empty directories in the slice passed in and logs
+// any errors copying the directories
+func copyEmptyDirectories(fsrc fs.Fs, fdst fs.Fs, entries fs.DirEntries) error {
+	if len(entries) == 0 {
+		return nil
+	}
+
+	var okCount int
+	for i := len(entries) - 1; i >= 0; i-- {
+		entry := entries[i]
+		dir, ok := entry.(fs.Directory)
+		if !ok {
+			fs.Errorf(fsrc, "Not a directory: %v", entry)
+			continue
+		}
+
+		items, err := fsrc.List(dir.Remote())
```

Comment:
I think calling `List` here will cause an extra API call for every directory which is probably not a lot better than trying to create the directories :-(